src/sentry/replays/tasks.py

```diff
@@ -0,0 +1,52 @@
+import time
+import uuid
+
+from django.conf import settings
+
+from sentry.replays.models import ReplayRecordingSegment
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, kafka_config
+from sentry.utils.pubsub import KafkaPublisher
+
+
+@instrumented_task(
+    name="sentry.replays.tasks.delete_recording_segments",
+    queue="default",
+    default_retry_delay=5,
+    max_retries=5,
+)
+def delete_recording_segments(project_id: int, replay_id: str, **kwargs: dict) -> None:
+    """Asynchronously delete a replay."""
+    _delete_replay_recording(project_id, replay_id)
+    _archive_replay(replay_id)
+
+
+def _delete_replay_recording(project_id: int, replay_id: str) -> None:
+    """Delete all recording-segments associated with a Replay."""
+    segments = ReplayRecordingSegment.objects.filter(
+        replay_id=replay_id, project_id=project_id
+    ).all()
+    for segment in segments:
+        segment.delete()  # Three queries + one request to the message broker
```

Comment:
does the comment "Three queries + one request to the message broker" mean it takes 3 SQL queries to delete a segment? and what does message broker mean in this context?

Reply:
Delete to File, delete to RecordingSegment, a task is scheduled to clean up blobs.  Not sure about the third query.  Might have been a typo.

Reply:
The other query is a read to FileBlob model.